### PR TITLE
Add architecture version field to internal doc

### PR DESCRIPTION
**Pull request by mjudeikis**
### Which issue this PR addresses:

Fixes prerequisite for 4.5

### What this PR does / why we need it:

This will help us to identify architecture differences when making changes, which will not be backward compatible (like NSG change we gonna do in 4.5)

### Test plan for issue:

manual testing

### Is there any documentation that needs to be updated for this PR?

No

### CODE REVIEW

File: pkg/cluster/install.go [hunk 1]
```diff
@@ -192,3 +194,14 @@ func (i *manager) addResourceProviderVersion(ctx context.Context) error {
 	})
 	return err
 }
+
+// addArchitectureVersion sets architecture version in
+// the cluster document for version-tracking purposes.
+func (i *manager) addArchitectureVersion(ctx context.Context) error {
+	var err error
+	i.doc, err = i.db.PatchWithLease(ctx, i.doc.Key, func(doc *api.OpenShiftClusterDocument) error {
+		doc.OpenShiftCluster.Properties.ArchitectureVersion = version.ArchitectureVersion
```
Comment (mjudeikis) [inline pkg/cluster/install.go:203]: This to match version.GitCommit so we keep all version-specific info in one place.

File: pkg/api/admin/openshiftcluster.go [hunk 2]
```diff
@@ -166,6 +167,15 @@ const (
...
 )
 
+// ArchitectureVersion represents an architecture version
+type ArchitectureVersion int
+
+// ArchitectureVersion constants
+const (
+	ArchitectureVersionV1 ArchitectureVersion = iota
+	ArchitectureVersionV2
+)
```
Comment (jim-minter) [inline pkg/api/admin/openshiftcluster.go:177]: Do we want to have to maintain this in multiple places?  Does an enum make it harder for us to do `if version < X` in our code, which we might need to do?

File: pkg/cluster/install.go [hunk 3]
```diff
@@ -42,6 +42,7 @@ func (i *manager) AdminUpgrade(ctx context.Context) error {
 		steps.Action(i.upgradeCertificates),
 		steps.Action(i.configureAPIServerCertificate),
 		steps.Action(i.configureIngressCertificate),
+		steps.Action(i.addArchitectureVersion),     // TODO(mj): Remove this once all cluster in production has this field populated
```
Comment (jim-minter) [inline pkg/cluster/install.go:45]: we can assume that our current clusters are at version 0 and omit this step.  In the 4.5 PR we can write a higher version at some point when we write i.doc

File: pkg/cluster/install.go (hunk 1 above)
Comment (jim-minter) [reply pkg/cluster/install.go:203]: could go in the 4.5 PR?

File: pkg/util/version/const.go [hunk 4]
```diff
@@ -1,9 +1,12 @@
 package version
 
+import "github.com/Azure/ARO-RP/pkg/api"
+
 // Copyright (c) Microsoft Corporation.
 // Licensed under the Apache License 2.0.
 
 var GitCommit = "unknown"
+var ArchitectureVersion = api.ArchitectureVersionV1
```
Comment (jim-minter) [inline pkg/util/version/const.go:9]: somewhere we will want to document what these mean

**Comment by mjudeikis** [comment]
@jim-minter are you suggesting something just like this;
```
// ArchitectureVersion represents an architecture version
type ArchitectureVersion int

// ArchitectureVersion constants
const (
	ArchitectureVersionV1 ArchitectureVersion = iota
	ArchitectureVersionV2
)

and start setting as `V2` hence int=1? so all existing clusters would be int=0?
```

**Pull request closed without merging**